src/app/components/authorization-page/authorization-page.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, ValidationErrors, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthorizationService } from '@services/authorization.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-authorization-page',
+  templateUrl: './authorization-page.component.html',
+  styleUrls: ['./authorization-page.component.scss']
+})
+export class AuthorizationPageComponent implements OnInit, OnDestroy {
+
+  public hidePassword = true;
+  public hideRepeatPassword = true;
+  public isRegistration = true;
+  public title = 'Registration';
+  public buttonText = 'Register';
+  public form!: FormGroup;
+  private _subscriptions = new Subscription();
+  private _emailPattern = '^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$';
+  private _passwordPattern = '^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*?[#?!@$%^&*-]).{8,}$';
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private router: Router,
+    private authService: AuthorizationService
+  ) {
+    if (this.router.url === '/authorization') {
+      this.isRegistration = false;
+      this.title = 'Log In';
+      this.buttonText = 'Login';
+    }
+  }
+
+  ngOnInit(): void {
+    this._buildForm();
+  }
+
+  public onSubmit(): void {
+    if (this.isRegistration) {
+      this._subscriptions.add(
+        this.authService.registerUser(this.form.value).subscribe(data => {
+          if (data.message === 'Success') {
+            this.router.navigate(['/authorization']);
+          }
+        })
+      );
+    } else {
+      this._subscriptions.add(
+        this.authService.loginUser(this.form.value).subscribe(data => {
+          if (data.message === 'Success') {
+            this.authService.storeUser(data);
+            this.router.navigate(['/home']);
+          }
+        })
+      );
+    }
+  }
+
+  private _checkPasswords(control: AbstractControl): ValidationErrors | null {
+    if (control.get('password')!.value !== control.get('repeatPassword')!.value) {
+      control.get('repeatPassword')!.setErrors({ notSame: true });
+    }
+    return null;
+  }
+
+  private _buildForm(): void {
+    if (this.isRegistration) {
+      this.form = this.formBuilder.group({
+        username: ['', [Validators.required, Validators.minLength(8), Validators.maxLength(20)]],
```

Comment:
@KovalivNataliia add pattern validator for username field. It must consist only characters of english alphabet.